test/testkit/mocks/backend/backend.go

```diff
@@ -1,59 +1,181 @@
 package backend
 
 import (
+	"fmt"
 	"path/filepath"
 
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	telemetryv1alpha1 "github.com/kyma-project/telemetry-manager/apis/telemetry/v1alpha1"
 	"github.com/kyma-project/telemetry-manager/test/testkit"
+	kitk8s "github.com/kyma-project/telemetry-manager/test/testkit/k8s"
+	kitmetric "github.com/kyma-project/telemetry-manager/test/testkit/kyma/telemetry/metric"
+	kittrace "github.com/kyma-project/telemetry-manager/test/testkit/kyma/telemetry/trace"
 	"github.com/kyma-project/telemetry-manager/test/testkit/mocks/backend/fluentd"
 )
 
+type SignalType string
+
+const (
+	// TelemetryDataFilename is the filename for the OpenTelemetry collector's file exporter.
+	TelemetryDataFilename = "otlp-data.jsonl"
+	defaultNamespaceName  = "default"
+
+	SignalTypeTraces  = "traces"
+	SignalTypeMetrics = "metrics"
+	SignalTypeLogs    = "logs"
+)
+
+type Option func(*Backend)
+
 type Backend struct {
-	name             string
-	namespace        string
-	exportedFilePath string
-	signalType       SignalType
-	withTLS          bool
-	certs            testkit.TLSCerts
+	name       string
+	namespace  string
+	signalType SignalType
+
+	PersistentHostSecret bool
+	WithTLS              bool
+	TLSCerts             testkit.TLSCerts
+
+	TracePipelineOptions  []kittrace.PipelineOption
+	MetricPipelineOptions []kitmetric.PipelineOption
+
+	ConfigMap        *ConfigMap
+	FluentDConfigMap *fluentd.ConfigMap
+	Deployment       *Deployment
+	ExternalService  *ExternalService
+	HostSecret       *kitk8s.Secret
 }
 
-func New(name, namespace, exportedFilePath string, signalType SignalType) *Backend {
-	return &Backend{
-		name:             name,
-		namespace:        namespace,
-		exportedFilePath: exportedFilePath,
-		signalType:       signalType,
-		withTLS:          false,
-		certs:            testkit.TLSCerts{},
+func New(name, namespace string, signalType SignalType, opts ...Option) (*Backend, error) {
```

Comment:
A tiny comment - let us not propagate the cert generate error here. We cannot really handle it gracefully and it must not happen. I think it would be even better to panic in the `build` function in this case